types/ember__object/test/octane.ts

```diff
@@ -109,9 +109,9 @@ class Bar extends EmberObject {
     @filter filterTest1: string[]; // $ExpectError
     @filter() filterTest2: string[]; // $ExpectError
     @filter('firstName') filterTest3: string[]; // $ExpectError
-    @filter('firstName', x => x) filterTest4: string[];
-    @filter('firstName', 'secondName', x => x) filterTest5: string[]; // $ExpectError
-    @filter('firstName', ['secondName'], x => x) filterTest6: string[];
+    @filter('firstName', Boolean) filterTest4: string[];
+    @filter('firstName', 'secondName', Boolean) filterTest5: string[]; // $ExpectError
+    @filter('firstName', ['secondName'], Boolean) filterTest6: string[];
```

Comment:
These make it really obvious that this is a breaking change. 😬 However, I’m already planning on doing the work to get a 4.x release done this week to prep for pulling these types into Ember mainline over the next couple quarters, so my current thinking (cc. @dfreeman @jamescdavis) is to basically pull these changes into *that* PR—I would just straight-up cherry-pick these commits into a new branch and go from there in terms of also cleaning up the other stuff we need to do for 4.x.

Reply:
Yeah, that would be good. I'll go ahead and close this PR. Thanks!